app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -482,6 +484,7 @@ def construct_repo_from_params(repo_params)
       root.upstream_username = repo_params[:upstream_username] if repo_params.key?(:upstream_username)
       root.upstream_password = repo_params[:upstream_password] if repo_params.key?(:upstream_password)
       root.ignorable_content = repo_params[:ignorable_content] if root.yum? && repo_params.key?(:ignorable_content)
+      root.ansible_collection_whitelist = repo_params[:ansible_collection_whitelist] if root.ansible_collection?
```

Comment:
As part of adding this new type, it'd be great to identify places where we can make the code more extensible, and hardcode type-specific entries less often.  Maybe lets get together and brainstorm and document some ways of doing it, even if we do it at a later time.

Reply:
In this case specifically, we could just ignore the check and let the field be set to nil?